frontend/components/sidebar/QueryList.tsx

```diff
@@ -217,20 +195,16 @@ function QueryList({ createQuery, show }: QueryListProps) {
       buttonLabel: 'Select Path',
       filters: [{ name: 'JSON', extensions: ['json'] }],
     };
-    const setFilePathCallback = (val) =>
-      queryDispatchContext!({
-        type: 'UPDATE_FILEPATH',
-        payload: val,
-      });
-    menuDispatch({
-      type: 'ASYNC_TRIGGER',
+    const setFilePathCallback = (val: string) =>
+      dispatch(updateFilePath(val));
+    dispatch(asyncTrigger({
```

Comment:
Replace current Context API code with correct redux function calls. You will need to import those functions from the slices